travel-agency-persistence/src/test/java/cz/fi/muni/pa165/travelagency/dao/ReservationDaoTest.java

```diff
@@ -0,0 +1,109 @@
+package cz.fi.muni.pa165.travelagency.dao;
+
+import cz.fi.muni.pa165.travelagency.EmbeddedDatabaseSpringConfig;
+import cz.fi.muni.pa165.travelagency.entity.Customer;
+import cz.fi.muni.pa165.travelagency.entity.Excursion;
+import cz.fi.muni.pa165.travelagency.entity.Reservation;
+import cz.fi.muni.pa165.travelagency.entity.Trip;
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.text.SimpleDateFormat;
+import java.util.HashSet;
+import java.util.Set;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+import org.springframework.test.context.support.DirtiesContextTestExecutionListener;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author Diana Vilkolakova
+ */
+@ContextConfiguration(locations = "/SpringXMLConfig.xml")
+@TestExecutionListeners(inheritListeners = false, listeners = {DependencyInjectionTestExecutionListener.class, DirtiesContextTestExecutionListener.class, TransactionalTestExecutionListener.class})
+@Transactional
+public class ReservationDaoTest extends AbstractTestNGSpringContextTests {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Autowired
+    private ReservationDao reservationDao;
+
+    private Reservation reservation;
+    private Customer customer;
+    private Trip trip1;
+    private Trip trip2;
+    private final BigDecimal price = new BigDecimal(5000);
+
+    @BeforeMethod
+    public void createReservations() {
+        reservation = new Reservation();
+        customer = new Customer();
+        
+        String dateFromFormat = "2000-11-01"; 
+        Date dateFrom = Date.valueOf(dateFromFormat);
+        dateFromFormat = "2015-11-01"; 
```

Comment:
Julo prisiel na este lepsi sposob, neviem ci to funguje, ale bolo by to lepsie:
Date date = new Date(2015, 1, 2);